// coffeelint-newline-at-eof/src/RuleProcessor.js

/*
Dreamscapes\coffeelint-newline-at-eof

Licensed under the BSD-3-Clause license
For full copyright and license information, please see the LICENSE file

@author     Robert Rossmann <rr.rossmann@me.com>
@copyright  2014 Robert Rossmann
@link       https://github.com/Dreamscapes/coffeelint-newline-at-eof
@license    http://choosealicense.com/licenses/BSD-3-Clause  BSD-3-Clause License
 */

(function() {
  var RuleProcessor;

  RuleProcessor = (function() {
    function RuleProcessor() {}

    RuleProcessor.prototype.rule = {
      name: 'newline_at_eof',
      description: 'Validates a newline policy at the end of each file',
      level: 'error',
      message: 'Incorrect end of file',
      behaviour: 'require'
    };

    RuleProcessor.prototype.lintLine = function(line, lineApi) {
      var behaviour;
      if (lineApi.isLastLine()) {
        behaviour = lineApi.config[this.rule.name].behaviour;
        if (behaviour === 'require' && line.length !== 0) {
          return {
            context: 'Expecting NEWLINE'
          };
        }
        if (behaviour === 'forbid' && line.length === 0) {
          return {
            context: 'Unexpected NEWLINE'
          };
        }
      } else {
        return null;
      }
    };

    return RuleProcessor;

  })();

  module.exports = RuleProcessor;

}).call(this);
